### PR TITLE
move wrap status update a level up

### PR DESCRIPTION
**Pull request by ranakan19**
move status update to include all errors from `ensureSpace`

### CODE REVIEW

File: controllers/usersignup/usersignup_controller_test.go [hunk 1]
```diff
@@ -4259,6 +4259,110 @@ func TestUserSignupStatusNotReady(t *testing.T) {
...
+	mur.Labels = map[string]string{
+		toolchainv1alpha1.MasterUserRecordOwnerLabelKey: userSignup.Name,
+		toolchainv1alpha1.UserSignupStateLabelKey:       "approved",
+	}
+
+	space := spacetest.NewSpace("john-doe",
+		spacetest.WithSpecTargetCluster("member-1"),
+		spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+		spacetest.WithFinalizer(),
+		spacetest.WithDeletionTimestamp())
```
Comment (mfrancisc) [inline controllers/usersignup/usersignup_controller_test.go:4286]: don't think it matters, but probably it's more clear that space is terminating ?
 
```suggestion
		spacetest.WithCondition(spacetest.Terminating()),
		spacetest.WithDeletionTimestamp())
```

File: controllers/usersignup/usersignup_controller_test.go [hunk 2]
```diff
@@ -4259,6 +4259,110 @@ func TestUserSignupStatusNotReady(t *testing.T) {
...
+		r, req, _ := prepareReconcile(t, userSignup.Name, ready, userSignup, commonconfig.NewToolchainConfigObjWithReset(t, testconfig.AutomaticApproval().Enabled(true)), baseNSTemplateTier, deactivate30Tier, mur, space)
+		InitializeCounters(t, NewToolchainStatus(
+			WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
+				"2,internal": 11, // 11 users signed-up 2 times, including our user above, even though she is not active at the moment
+				"3,internal": 10, // 10 users signed-up 3 times
+			}),
+			WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
+				string(metrics.Internal): 21,
+			}),
+		))
```
Comment (mfrancisc) [inline controllers/usersignup/usersignup_controller_test.go:4328]:
```suggestion
```

minor thing - I don't think the counters are required unless checked at the end of the test

**Review by mfrancisc — APPROVED** [review]
Looks good 👍 

Left only few minor suggestions

File: controllers/usersignup/usersignup_controller_test.go (hunk 2 above)
Comment (ranakan19) [reply controllers/usersignup/usersignup_controller_test.go]: yep, thanks for catching that. I was getting some errors before if i did not initialize counters, but it was a side effect of copying it from other test case.
fixed in https://github.com/codeready-toolchain/host-operator/pull/744/commits/10d70b84af0481524da6afd097097ace21e77543

**Review by alexeykazakov — APPROVED** [review]
Looks good!

File: controllers/usersignup/usersignup_controller_test.go (hunk 2 above)
Comment (mfrancisc) [reply controllers/usersignup/usersignup_controller_test.go]: No worries and thanks for addressing my comments 👍

**Pull request merged**